superduper/jobs/queue.py

```diff
@@ -64,9 +65,14 @@ class BaseQueuePublisher(ABC):
     """
 
     def __init__(self, uri: t.Optional[str]):
-        self.queue: t.Dict = {}
-        self.components: t.Dict = {}
         self.uri: t.Optional[str] = uri
+        self.queue: t.Dict = defaultdict(lambda: [])
+
+        from superduper.misc.special_dicts import ArgumentDefaultDict
+
+        self.components = ArgumentDefaultDict(
```

Comment:
Without this it doesn't work if you reload the system.

Reply:
I am not sure if we can use `defaultdict(lambda x: self.db.load(*x))`